internal/factsengine/gatherers/hostsfile.go

```diff
@@ -128,11 +128,22 @@ func hostsFileToMap(lines []string) (map[string][]string, error) {
 			}
 		}
 		hostnames := strings.Fields(paramsMap["hostnames"])
+
 		for _, hostname := range hostnames {
-			hostsFileMap[hostname] = append(hostsFileMap[hostname], paramsMap["ip"])
+			if i, found := hostsFileMap[hostname]; found {
```

Comment:
just a minor thing, I really appreciate the fact that you are using proper variable name, it's all more readable, maybe we could use ip, instead of i, seems more aligned with the other names, what do you think?